txscript/script.go

```diff
@@ -104,11 +111,22 @@ func isStakeScriptHashScript(script []byte) bool {
 		script[23] == OP_EQUAL
 }
 
+// isTreasuryScriptHashScript returns whether or not the passed script is a
+// stake-tagged pay-to-script-hash script.
+func isTreasuryScriptHashScript(script []byte) bool {
+	return len(script) == 24 &&
+		isTreasuryOpcode(script[0]) &&
+		script[1] == OP_HASH160 &&
+		script[2] == OP_DATA_20 &&
+		script[23] == OP_EQUAL
+}
+
 // isAnyKindOfScriptHash returns whether or not the passed script is either a
 // regular pay-to-script-hash script or a stake-tagged pay-to-script-hash
 // script.
 func isAnyKindOfScriptHash(script []byte) bool {
-	return isScriptHashScript(script) || isStakeScriptHashScript(script)
+	return isScriptHashScript(script) || isStakeScriptHashScript(script) ||
+		isTreasuryScriptHashScript(script)
```

Comment:
This is a consensus breaking change.  It needs to be gated behind the `ScriptVerifyTreasury` flag.